test/integration_test.py

```diff
@@ -1378,14 +1378,35 @@ def source_id_from_hit(hit: JSON) -> str:
         bundle_type = self._bundle_type(catalog)
         project_type = self._project_type(catalog)
 
-        hits = self._get_entities(catalog, project_type)
-        sources_found = set()
-        for hit in hits:
-            source_id = source_id_from_hit(hit)
-            sources_found.add(source_id)
-            self.assertEqual(source_id not in managed_access_source_ids,
-                             one(hit[project_type])['accessible'])
-        self.assertIsSubset(managed_access_source_ids, sources_found)
+        # The "None" case must the final iteration since it determines the
+        # return value of this method.
+        for accessible in True, False, None:
+            filters = None if accessible is None else {
+                special_fields.accessible: {'is': [accessible]}
+            }
+            # If we were to move this assignment to inside the subtest, this
+            # method would return an incorrect value (from a previous iteration)
+            # if the assignment were to raise an exception on the final
+            # iteration.
```

Comment:
I do believe that there is a way to handle this more elegantly using control flow or an added assertion outside the subtest.